addons/web_editor/static/tests/link_test.js

```diff
@@ -0,0 +1,458 @@
+            const { editor, editable } = onMount();
+            const node = editable.querySelector("p");
+            setSelection(node, 0);
+            insertText(editor, "Hello");
+            triggerEvent(node, "keydown", { key: "a", ctrlKey: true });
+            await nextTick();
+            editor.document.querySelector(".fa-link").click();
+            await nextTick();
+            inputText('input[id="o_link_dialog_url_input"]', "#");
+            editor.document.querySelector(".o_dialog footer button.btn-primary").click();
+            await nextTick();
+            assert.strictEqual(
+                editable.innerHTML,
+                `<p><a href="#" target="_blank">Hello</a></p>`
+            );
+        });
+
+        QUnit.test("should set the link on two existing characters", async function (assert) {
+            const { editor, editable } = onMount();
+            const node = editable.querySelector("p");
+            setSelection(node, 0);
+            insertText(editor, "Hello");
+            setSelection(node, 1, node, 3);
+            editor.document.querySelector(".fa-link").click();
+            await nextTick();
+            inputText('input[id="o_link_dialog_url_input"]', "#");
+            editor.document.querySelector(".o_dialog footer button.btn-primary").click();
+            await nextTick();
+            assert.strictEqual(
+                editable.innerHTML,
+                `<p>H<a href="#" target="_blank">el</a>lo</p>`
+            );
+        });
+
+        QUnit.test("Should be able to insert link on empty p", async function (assert) {
+            const { editor, editable } = onMount();
+            const node = editable.querySelector("p");
+            setSelection(node, 0);
+            triggerEvent(node, "keydown", { key: "a", ctrlKey: true });
+            await nextTick();
+            editor.document.querySelector(".fa-link").click();
+            await nextTick();
+            inputText('input[id="o_link_dialog_url_input"]', "#");
+            editor.document.querySelector(".o_dialog footer button.btn-primary").click();
+            await nextTick();
+            assert.strictEqual(
+                editable.innerHTML,
+                `<p><a href="#" target="_blank"><br></a></p>`
+            );
+        });
+
+
+        QUnit.module("PowerBox related");
+
+        QUnit.test("should insert a link and preserve spacing", async function (assert) {
+            const { editor, editable } = onMount();
+            const node = editable.querySelector("p");
+            setSelection(node, 0);
+            insertText(editor, "ab");
+            setSelection(node.firstChild, 1, node.firstChild, 1);
+            insertText(editor, " ");
+            setSelection(node.firstChild, 2, node.firstChild, 2);
+            insertText(editor, " ");
+            setSelection(node.firstChild, 3, node.firstChild, 3);
+            insertText(editor, " ");
```

Comment:
`insertText(editor, "a   b")` did not work?

Reply:
yes,some how the space between a and b gets  trimmed,
![image](https://github.com/odoo/odoo/assets/103405229/44eccdf7-8999-48ad-801e-1148406b4bed)